### PR TITLE
chore: clean up unused artifacts, add deprecated notices.

### PR DESCRIPTION
**Pull request by HumairAK**
1. Remove crd viewer 
2. Add deprecated warnings for v1 specific fields in dspa 
3. Ignore linting on dspa_params.go, so ci does not fail when it sees statticheck deprecated errors 
4. Update dspa_all_fields.yaml to reflect latest state of possible dspa configuration options

### CODE REVIEW

**Comment by dsp-developers** [comment]
A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-610`
An OCP cluster where you are logged in as cluster admin is required.

To use this image run the following:

```bash
cd $(mktemp -d)
git clone git@github.com:opendatahub-io/data-science-pipelines-operator.git
cd data-science-pipelines-operator/
git fetch origin pull/610/head
git checkout -b pullrequest 69fd88e012a02077bcb3a9e4d76ffc65dbe76d80
oc new-project opendatahub
make deploy IMG="quay.io/opendatahub/data-science-pipelines-operator:pr-610"
```

More instructions [here](https://github.com/opendatahub-io/data-science-pipelines-operator#deploy-dsp-instance) on how to deploy and test a Data Science Pipelines Application.

File: api/v1alpha1/dspipeline_types.go [hunk 1]
```diff
@@ -42,9 +42,6 @@ type DSPASpec struct {
 	*ObjectStorage `json:"objectStorage"`
 	*MLMD          `json:"mlmd,omitempty"`
 	// +kubebuilder:validation:Optional
-	// +kubebuilder:default:={deploy: false}
```
Comment (gregsheremeta) [inline api/v1alpha1/dspipeline_types.go:45]: no deprecation period first?

File: api/v1alpha1/dspipeline_types.go [hunk 2]
```diff
@@ -59,69 +56,83 @@ type APIServer struct {
...
+	// server pod to trust this connection. CA Bundle should be provided
+	// as values within configmaps, mapped to keys.
+	CABundle *CABundle `json:"cABundle,omitempty"`
+
+	// CustomServerConfig is a custom config file that you can provide
+	// for the api server to use instead.
+	CustomServerConfig *ScriptConfigMap `json:"customServerConfigMap,omitempty"`
+
 	// Default: true
+	// Deprecated: v1 only, will be removed in the future.
```
Comment (gregsheremeta) [inline api/v1alpha1/dspipeline_types.go:84]: hm, should we be bumping our API to v2alpha1 for DSP v2?

Comment (gregsheremeta) [reply api/v1alpha1/dspipeline_types.go:84]: also, depending on the answer to the above, we may want to say "v1alpha1" here. I think you mean "DSP v1" when you say "v1", but in this CRD API file, "v1" means a kubernetes API and not a version of the project. It could cause confusion.

File: config/samples/v2/dspa-all-fields/dspa_all_fields.yaml
Comment (gregsheremeta) [inline config/samples/v2/dspa-all-fields/dspa_all_fields.yaml:1]: seems not related -- should be its own PR

Comment (HumairAK) [reply config/samples/v2/dspa-all-fields/dspa_all_fields.yaml:1]: removing unused artifacts and bringing v2 crd api up to speed is the goal of the pr

File: api/v1alpha1/dspipeline_types.go (hunk 1 above)
Comment (HumairAK) [reply api/v1alpha1/dspipeline_types.go:45]: this was never part of any release, so we don't need this

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-610`

File: api/v1alpha1/dspipeline_types.go (hunk 2 above)
Comment (HumairAK) [reply api/v1alpha1/dspipeline_types.go]: > hm, should we be bumping our API to v2alpha1 for DSP v2?

This is something we should have probably considered way ahead of time. I think crd version conversion strategy is upto us to decide when we do it. I'm not sure what the standard practice is here, but I don't think we should update the version of the crd this close to release, it would be difficult to account for all edge cases we need to handle. 

I updated v1 to say "DSP V1" however.

File: config/samples/v2/dspa-all-fields/dspa_all_fields.yaml [hunk 3]
```diff
@@ -12,13 +12,29 @@ spec:
...
+    # requires this configmap to be created beforehand,
+    customServerConfigMap:
+      name: configmapname
+      key: keyname
+    # the following are v1 specific options in spec.apiServer.*
     applyTektonCustomResource: true
     archiveLogs: false
-    artifactImage: quay.io/modh/odh-ml-pipelines-artifact-manager-container:v1.18.0-8
-    cacheImage: registry.access.redhat.com/ubi8/ubi-minimal
+    artifactImage: uay.io/opendatahub/ds-pipelines-artifact-manager:latest
```
Comment (amadhusu) [inline config/samples/v2/dspa-all-fields/dspa_all_fields.yaml:36]: Typo -> q missing in `quay.io`

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-610`

**Review by DharmitD — APPROVED** [review]
Changes lgtm, and since the CI checks also pass

/Approve
